common/namespace/replicationTaskExecutor.go

```diff
@@ -164,11 +164,12 @@ func (h *namespaceReplicationTaskExecutorImpl) handleNamespaceCreationReplicatio
 				Data:        task.Info.Data,
 			},
 			Config: &persistencespb.NamespaceConfig{
-				Retention:               task.Config.GetWorkflowExecutionRetentionTtl(),
-				HistoryArchivalState:    task.Config.GetHistoryArchivalState(),
-				HistoryArchivalUri:      task.Config.GetHistoryArchivalUri(),
-				VisibilityArchivalState: task.Config.GetVisibilityArchivalState(),
-				VisibilityArchivalUri:   task.Config.GetVisibilityArchivalUri(),
+				Retention:                    task.Config.GetWorkflowExecutionRetentionTtl(),
+				HistoryArchivalState:         task.Config.GetHistoryArchivalState(),
+				HistoryArchivalUri:           task.Config.GetHistoryArchivalUri(),
+				VisibilityArchivalState:      task.Config.GetVisibilityArchivalState(),
+				VisibilityArchivalUri:        task.Config.GetVisibilityArchivalUri(),
+				CustomSearchAttributeAliases: task.Config.GetCustomSearchAttributeAliases(),
```

Comment:
Thanks fixing this!
But I wonder if there's a way to avoid this issue in the future?
I feel like every time a new field is added, it's prone have this same issue.

Reply:
Reflection? Do we need to do this deep copy at all? 
At least we should be able to add test which check all fields using reflection.